packages/SwingSet/src/kernel/kernel.js

```diff
@@ -586,6 +589,7 @@ export default function buildKernel(
     }
 
     function sendResponse(args) {
+      // @ts-ignore see assert(...) above
```

Comment:
IMO `// @ts-ignore` is dangerous in these type of cases as it doesn't target the exact argument for which we want to ignore the type error.

In general there are 2 alternative options:
- Assign to a temporary const that has the right type, and `ts-ignore` that assignment.
- cast the argument at the site of usage

While I hate changing runtime code to satisfy the type system, the ugliness of casting (and the need to often cast to `unknown` if the types really are incompatible) often makes assigning to another `const` the path of least resistance.

In this specific case, it is very unfortunate that TypeScript doesn't support an equivalent to the [non-null assertion operator (`!`) in jsdoc](https://github.com/microsoft/TypeScript/issues/23405), but the type casting shouldn't require a  trip to `unknown`.

However given the `assert`, there may be a way to make this work since TypeScript now supports type definition of assert functions: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

Reply:
Well interestingly `assert` is already defined correctly: https://github.com/Agoric/agoric-sdk/blob/8eba7f81fdd2e91f76ecb08ec2059961b2d05c5c/packages/assert/src/types.js#L19

Wondering what went wrong that tsc lost the type narrowing.

Reply:
> Well interestingly `assert` is already defined correctly:

Thanks to @michaelfig . I just slavishly imitated his typing patterns.

Reply:
Yeah and I reproduced the pattern of nested closures, and tsc seem to do the right thing: [Playground](https://www.typescriptlang.org/play?jsx=0&strict=true#code/GYVwdgxgLglg9mABAQwM6oKYCcoAoIIAmMsCAXCmAJ4A0iAtqgOYD8FqUWMYTAlBWkw5UiAmGKkkAbwBQiRDGCJcAQjET4YXolnz5UABZY4Ad0RgMZgKJZjWXIz4BuOYgC+MjzLEcUFAPIARgBWGNCIAD7mIAA2MYgAvDpuLjKgkJKIwHBwuNq6eoLYeMi8qXqIWBhQIFhIeYkAfDquFfI+UIiBiSgubXo+cDEYAHQxcEy4gWWt7p5AA)